zcash_primitives/src/transaction/builder.rs

```diff
@@ -50,157 +45,36 @@ use crate::sapling::prover::mock::MockTxProver;
 
 const DEFAULT_TX_EXPIRY_DELTA: u32 = 20;
 
-/// If there are any shielded inputs, always have at least two shielded outputs, padding
-/// with dummy outputs if necessary. See <https://github.com/zcash/zcash/issues/3615>.
-const MIN_SHIELDED_OUTPUTS: usize = 2;
-
 #[derive(Debug, PartialEq)]
 pub enum Error {
-    AnchorMismatch,
-    BindingSig,
     ChangeIsNegative(Amount),
     InvalidAddress,
     InvalidAmount,
     NoChangeAddress,
-    SpendProof,
     TzeWitnessModeMismatch(u32, u32),
     TransparentBuildError(transparent::Error),
+    SaplingBuildError(sapling::Error),
```

Comment:
Avoid duplicative suffixes on enum cases:
```suggestion
    SaplingBuild(sapling::Error),
```